lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php

```diff
@@ -34,7 +34,11 @@
  * @link   www.doctrine-project.org
  * @since  2.5
  */
-abstract class AbstractMySQLDriver implements Driver, ExceptionConverterDriver, VersionAwarePlatformDriver
+abstract class AbstractMySQLDriver implements
+    Driver,
+    ExceptionConverterDriver,
+    VersionAwarePlatformDriver,
+    ForeignKeyCheckDeactivatableDriver
```

Comment:
This is something that we really shouldn't expose to the outside world, as it opens also a series of cans of worms.

I would rather have the intent of the actual affected operations to be either correctly have workarounds built into DBAL (truncation, in this case), or just not using DBAL facilities at all (since we're going into veeeeeery specific details of horrible hacks like MySQL's)